vertical-pod-autoscaler/pkg/recommender/routines/recommender_routines_test.go

```diff
@@ -0,0 +1,267 @@
+		})
+}
+
+func getTestVPAObj(cluster *model.ClusterState, container model.ContainerID) *model.Vpa {
+	vpa := test.VerticalPodAutoscaler().WithName(testVpaID.VpaName).
+		WithNamespace(testVpaID.Namespace).WithContainer(container.ContainerName).WithUpdateMode(testUpdateModeAuto).Get()
+
+	labelSelector, _ := metav1.ParseToLabelSelector(testSelectorStr)
+	parsedSelector, _ := metav1.LabelSelectorAsSelector(labelSelector)
+	err := cluster.AddOrUpdateVpa(vpa, parsedSelector)
+	if err != nil {
+		klog.Fatalf("AddOrUpdateVpa() failed: %v", err)
+	}
+
+	return cluster.Vpas[testVpaID]
+}
+
+// Verifies that recommendation does not occur and remains Nil
+// if only memory samples are added onto the aggregate container state
+// histogram. The variable TotalSamplesCount is counted only for CPU
+// samples and a similar for memory should be introduced.
+func TestRecommendationForZeroTotalSamplesCount(t *testing.T) {
+	cluster := model.NewClusterState()
+	cluster.AddOrUpdatePod(testPodID1, testLabels, apiv1.PodRunning)
+	container := model.ContainerID{PodID: testPodID1, ContainerName: "app-A"}
+
+	assert.NoError(t, cluster.AddOrUpdateContainer(container, testRequest))
+	assert.NoError(t, addTestMemorySample(cluster, container, 2e9, testTimestamp))
+
+	pod := cluster.Pods[testPodID1]
+	aggregateStateKey := cluster.MakeAggregateStateKey(pod, "app-A")
+	aggregateResources := cluster.GetAggregateStateMap()
+	// Note: This returns TotalSamplesCount as 0 as we are only adding memory sample
+	// This is handled in another test case (TestTotalSamplesCountMemorySamplesOnly).
+	// Ignorning and continuing to seek recommendations for now. Fix it to equal to 1.
+	assert.Equal(t, 0, aggregateResources[aggregateStateKey].TotalSamplesCount)
+
+	recommender := logic.CreatePodResourceRecommender()
+	vpaObj := getTestVPAObj(cluster, container)
+	recommendedResources := recommender.GetRecommendedPodResources(GetContainerNameToAggregateStateMap(vpaObj))
+	assert.NotEmpty(t, recommendedResources)
+
+}
+
+// Verifies the number of samples it takes for the current
+// VPA recommender estimation to beat the current surge in usage.
+// The surge is determined and samples w.r.t that is added using
+// model.OOMBumpUpRatio
+func TestAddMemorySurgeWithOOMBumpUpRatio(t *testing.T) {
+	config := model.GetAggregationsConfig()
+	maxAllowedMemory := 5e10
+	cpuSamples := []float64{1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0}
+	cpuHistogram := addTestCPUHistogram(cpuSamples, 1.0, testTimestamp, config.CPUHistogramOptions)
+
+	var memSamples []float64
+	for i := 0; i < 20; i++ {
+		memSamples = append(memSamples, 1e6)
+	}
+	memoryPeaksHistogram := addTestMemoryHistogram(memSamples, 1.0, testTimestamp, config.MemoryHistogramOptions)
+
+	ResourcePercentile := 0.9 // Same as in podResourceRecommender for target estimator
+	resourceEstimation := GetPercentileResourceEstimation(ResourcePercentile, ResourcePercentile,
+		cpuHistogram, memoryPeaksHistogram)
```

Comment:
Can't we call [podResourceRecommender.GetRecommendedPodResources](https://github.com/gardener/autoscaler/blob/a842944a648b7e3d87b2915f4c5fea077db9969b/vertical-pod-autoscaler/pkg/recommender/logic/recommender.go#L56:34) instead?

Reply:
ACK. Removed as not using estimator but using GetRecommendedPodResources now.